components/places/src/types.rs

```diff
@@ -17,7 +18,20 @@ impl SyncGuid {
     pub fn new() -> Self {
         SyncGuid(sync15::random_guid().unwrap())
     }
+
+    pub fn as_root(&self) -> Option<BookmarkRootGuid> {
+        BookmarkRootGuid::from_str(&self.0)
+    }
+
+    pub fn is_root(&self) -> bool {
+        BookmarkRootGuid::from_str(&self.0).is_some()
+    }
+
+    pub fn is_specific_root(&self, root: BookmarkRootGuid) -> bool {
```

Comment:
Why not implement `PartialEq<BookmarkRootGuid> for SyncGuid`?

Reply:
I do, actually, I forgot that existed.